lib/stat_tracker.rb

```diff
@@ -143,6 +143,134 @@ def average_goals_by_season
     avg_season_goals
   end
 
+  def count_of_teams
+    team_count = 0
+    contents = CSV.open(@team_path, headers: true, header_converters: :symbol)
+    contents.each do |row|
+      teams = row[:team_id].to_i
+      if teams != 0
+        team_count += 1
+      end
+    end
+    team_count
+  end
+
+  def best_offense
+    game_teams_csv = CSV.open(@game_teams_path, headers: true, header_converters: :symbol)
+    goals_by_team_id = Hash.new(0)
+    game_count_by_team_id = Hash.new(0)
+    game_teams_csv.each do |row|
+      goals_by_team_id[row[:team_id].to_i] += row[:goals].to_i
+      game_count_by_team_id[row[:team_id].to_i] += 1
+    end
+    average_teamid_score = goals_by_team_id.map{|team_id, goals| [team_id, goals.to_f / game_count_by_team_id[team_id]] }.to_h
+    best_offense_team_id = average_teamid_score.max_by { |team_id, average_goals| average_goals  }[0]
+    teams_csv = CSV.open(@team_path, headers: true, header_converters: :symbol)
+    team_row = teams_csv.find do |row|
+      row[:team_id].to_i == best_offense_team_id
+    end
+    team_row[:teamname]
```

Comment:
great start with the method. i did write a team_names method that returns a hash {key:team_id, value:team name} so we can easily translate between the two. and use this for all other methods that need a team name and keep our code DRY